src/deployment/mod.rs

```diff
@@ -0,0 +1,1084 @@
+        base_dir.pop();
+        if !base_dir.exists() {
+            let _ = std::fs::create_dir(base_dir);
+        }
+    }
+
+    let file = deployment.to_specification_file();
+
+    let content = match serde_yaml::to_string(&file) {
+        Ok(res) => res,
+        Err(err) => return Err(format!("unable to serialize deployment {}", err)),
+    };
+
+    let mut file = match File::create(&target_path) {
+        Ok(file) => file,
+        Err(e) => {
+            return Err(format!(
+                "unable to create file {}: {}",
+                target_path.display(),
+                e
+            ));
+        }
+    };
+    match file.write_all(content.as_bytes()) {
+        Ok(_) => (),
+        Err(e) => {
+            return Err(format!(
+                "unable to write file {}: {}",
+                target_path.display(),
+                e
+            ));
+        }
+    };
+    Ok(())
+}
+
+pub fn generate_default_deployment(
+    manifest_path: &PathBuf,
+    network: &Option<StacksNetwork>,
+) -> Result<(DeploymentSpecification, DeploymentGenerationArtifacts), String> {
+    let mut project_config = ProjectManifest::from_path(&manifest_path);
+    let chain_config = ChainConfig::from_manifest_path(&manifest_path, &network);
+
+    let node = match network {
+        None => None,
+        Some(StacksNetwork::Devnet) => Some(
+            chain_config
+                .network
+                .node_rpc_address
+                .unwrap_or("http://localhost:20443".to_string()),
+        ),
+        Some(StacksNetwork::Testnet) => Some(
+            chain_config
+                .network
+                .node_rpc_address
+                .unwrap_or("http://stacks-node-api.testnet.stacks.co".to_string()),
+        ),
+        Some(StacksNetwork::Mainnet) => Some(
+            chain_config
+                .network
+                .node_rpc_address
+                .unwrap_or("http://stacks-node-api.mainnet.stacks.co".to_string()),
+        ),
+    };
+
+    let deployment_fee_rate = chain_config.network.deployment_fee_rate;
+
+    let default_deployer = match chain_config.accounts.get("deployer") {
+        Some(deployer) => deployer,
+        None => {
+            return Err(format!(
+                "{} unable to retrieve default deployer account",
+                red!("x")
+            ));
+        }
+    };
+
+    let mut transactions = vec![];
+    let mut contracts_map = BTreeMap::new();
+    let mut requirements_asts = BTreeMap::new();
+    let mut requirements_deps = HashMap::new();
+
+    let mut settings = SessionSettings::default();
+    let parser_version = project_config.repl_settings.parser_version;
+    settings.repl_settings = project_config.repl_settings.clone();
+    let session = Session::new(settings.clone());
+    let mut boot_contracts_asts = session.get_boot_contracts_asts();
+    requirements_asts.append(&mut boot_contracts_asts);
+
+    // Only handle requirements in test environments
```

Comment:
I'll update the comment. In the context of Devnet/Testnet/Mainnet, we won't be deploying the requirements, but we still need to build the ASTs / dependency graphs, so that we can "resolve" the actual protocol. We will simply avoid creating transactions for this requirements in the final deployment specification.

Reply:
Addressed with latest